### PR TITLE
Fix NPE when listing buckets in empty project and blobs in empty bucket

### LINKED ISSUE #174: NPE From some Storage methods
For example the following calls when listing a bucket that is empty or getting an item from a bucket that exists but the item does not:
```java
    storage.list( bucketName )
    storage.get( bucketName, itemName )
```
The NPE is generated from the Preconditions class called via transform() in the Iterables class:
```java
public static <T> T checkNotNull(T reference) {
        if(reference == null) {
            throw new NullPointerException();
        } else {
            return reference;
        }
    }
```
It would be nicer to return a null to the caller perhaps.  wdyt?

### PR DESCRIPTION
**Pull request by mziccard**
This PR fixes #174.

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java [hunk 1]
```diff
@@ -259,14 +262,17 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket bucketPb) {
...
+        result.y() == null ? ImmutableList.<BlobInfo>of() : Iterables.transform(result.y(),
             new Function<StorageObject, BlobInfo>() {
               @Override
               public BlobInfo apply(StorageObject storageObject) {
                 return BlobInfo.fromPb(storageObject);
               }
-            }));
+            });
+    return new BaseListResult<>(new BlobPageFetcher(bucket, serviceOptions, cursor, optionsMap),
+        cursor,
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java]: I don't think line wrapping makes it more readable in this case. do you?

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java [hunk 2]
```diff
@@ -259,14 +262,17 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket bucketPb) {
           }
         }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
     String cursor = result.x();
-    return new BaseListResult<>(new BlobPageFetcher(bucket, serviceOptions, cursor, optionsMap), cursor,
-        Iterables.transform(result.y(),
+    Iterable<BlobInfo> blobs =
+        result.y() == null ? ImmutableList.<BlobInfo>of() : Iterables.transform(result.y(),
             new Function<StorageObject, BlobInfo>() {
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java]: Fine as is. Another way (insignificantly less efficient and arguable more readable could have been just to change `result.y()` to `MoreObjects.firstNonNull(result.y(), ImmutableList.of())`, ....

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java [hunk 3]
```diff
@@ -234,14 +235,16 @@ public StorageObject call() {
           }
         }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
     String cursor = result.x();
-    return new BaseListResult<>(new BucketPageFetcher(serviceOptions, cursor, optionsMap), cursor,
-        Iterables.transform(result.y(),
+    Iterable<BucketInfo> buckets =
+        result.y() == null ? ImmutableList.<BucketInfo>of() : Iterables.transform(result.y(),
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java]: So `storage.buckets()` or `objects.list("bucket")` can return `null`?

Adding a test case?

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java]: It seems slightly more readable to me :)

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java]: Yes and yes. I am writing tests for the whole `StorageImpl` class but I have not finished yet, I did not want to push a WIP with this PR.

**Pull request merged**